### DISCUSSION TITLE
Line graph of cumulative sums over time aggregated by categorical value

### DISCUSSION THREAD

**Discussion by jimmywan** [vega/altair#2535] [Q&A] [answered]
I generated this using a bunch of custom pandas pivot/melt/cumsum logic:
![image](https://user-images.githubusercontent.com/10406673/146076069-308b8475-cd5a-477c-be9a-24500af68e41.png)

I'd like to use altair transforms instead, but I don't quite understand how to do so.

I tried repurposing [this](https://altair-viz.github.io/gallery/cumulative_count_chart.html#) but got lost. Is there another/better place to look for docs/examples of what I'm trying to accomplish? Happy to try and contribute some update examples if that would be helpful.

**Comment by jakevdp** [comment, marked as answer]
It should look pretty much exactly like the example you linked to, except with a `groupby` in the window transform and an additional `color` encoding. Here's a short example:
```python
import pandas as pd
import altair as alt

data = pd.DataFrame({
  'qtr': ['2016Q1', '2016Q2', '2016Q3', '2016Q4', '2017Q1', '2017Q2', '2017Q3', '2017Q4',
          '2016Q1', '2016Q2', '2016Q3', '2016Q4', '2017Q1', '2017Q2', '2017Q3', '2017Q4'],
  'company': ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A',
              'B', 'B', 'B', 'B', 'B', 'B', 'B', 'B'],
  'value': [5, 4, 5, 6, 4, 7, 8, 3,
            7, 8, 8, 9, 6, 5, 8, 7],
})

alt.Chart(data).transform_window(
    cumulative='sum(value)',
    sort=[{"field": "qtr"}],
    groupby=['company'],
).mark_line().encode(
    x='qtr:N',
    y='cumulative:Q',
    color='company:N',
)
```
![visualization (85)](https://user-images.githubusercontent.com/781659/146084882-ac9c7f53-4b80-4d95-8d9a-b6942ff3479b.png)

**Reply by jimmywan** [reply]
Apologies if I didn't make it clear in the ask, I'm trying to plot multiple lines, one for each categorical value.

**Reply by jakevdp** [reply]
Ah, sorry I missed that. Answer updated.

**Reply by jimmywan** [reply]
tyvm!
